### PR TITLE
CodeQL Warnings Fixes

### PR DESCRIPTION
**Pull request by hugoghx**

### CODE REVIEW

**Comment by hugoghx** [comment]
Is it possible to tell the backport functionality to not merge the backport PR automatically after this PR is merged? I'm not entirely sure all of these changes will go cleanly into all release branches

**Comment by hugoghx** [comment]
TODO: Manually backport to `release/0.17.x` and `release/0.16.x` where applicable

File: internal/cmd/base/base.go
Comment (johanbrandhorst) [inline internal/cmd/base/base.go:119]: Can we make this a uint16? Ports on systems are limited to this number already.

File: internal/cmd/base/servers.go
Comment (johanbrandhorst) [inline internal/cmd/base/servers.go:125]: uint16?

**Review by johanbrandhorst — COMMENTED** [review]
Looks great, just wondering if we can use 16 bit unsigned integers directly for ports.

File: internal/cmd/base/base.go
Comment (louisruch) [reply internal/cmd/base/base.go:119]: +1

**Review by louisruch — APPROVED** [review]

File: internal/cmd/base/base.go
Comment (hugoghx) [reply internal/cmd/base/base.go:119]: The only blocker to changing this to `uint16` is that we then couldn't pass a pointer to this variable to all the `f.Uint64Var` calls we do.

File: internal/cmd/base/servers.go
Comment (hugoghx) [reply internal/cmd/base/servers.go:125]: The only blocker to changing this to `uint16` is that we then couldn't pass a pointer to this variable to all the `f.Uint64Var` calls we do.

File: internal/cmd/base/base.go
Comment (hugoghx) [reply internal/cmd/base/base.go:119]: Yeah, Louis and I were looking at the possibility of adding that type in, should be simple 👍

Comment (hugoghx) [reply internal/cmd/base/base.go]: All done - See [df25e75](https://github.com/hashicorp/boundary/pull/5027/commits/df25e75990daedd9b3b6620f6bd9eb8e8855d011) and https://github.com/hashicorp/boundary/pull/5027/commits/b0880a8fd941ca3db45cad5273d15c88bc0730eb

File: internal/cmd/base/servers.go
Comment (hugoghx) [reply internal/cmd/base/servers.go]: All done - See [df25e75](https://github.com/hashicorp/boundary/pull/5027/commits/df25e75990daedd9b3b6620f6bd9eb8e8855d011) and https://github.com/hashicorp/boundary/pull/5027/commits/b0880a8fd941ca3db45cad5273d15c88bc0730eb

**Review by johanbrandhorst — APPROVED** [review]
Looks great, thank you!!

**Comment by hugoghx** [comment]
PS: I've merged this branch (https://github.com/hashicorp/boundary/commit/1c62470ba6ee3eb54fb4309dc73f34e8eb7190c0) but it looks like this GitHub PR didn't get the memo 🤔

**Comment by hugoghx** [comment]
# **This has been merged**

I tried an empty commit to refresh the PR to see if it'd get marked as merged, but it did not. Closing the PR.
Merged into Boundary `main` at https://github.com/hashicorp/boundary/commit/1c62470ba6ee3eb54fb4309dc73f34e8eb7190c0

**Pull request closed without merging**
